src/utils/index.ts

```diff
@@ -368,3 +325,26 @@ export const discordInvites = {
   review_guidelines: "https://discord.gg/jqj4maCs8p",
   feedback: "https://discord.gg/W4cmWRhMnr",
 };
+
+export function parseJsonArray<T>(possibleArray: unknown): {
+  ok: boolean;
+  data: T | null;
+} {
+  try {
+    const data = JSON.parse(possibleArray as string);
+    return { ok: Array.isArray(data), data };
+  } catch (e) {
+    return { ok: false, data: null };
+  }
+}
+
+export function compareUrls(url1: URL, url2: URL) {
+  const host1 = url1.host.replace("www.", "");
+  const host2 = url2.host.replace("www.", "");
+
+  return (
+    host1 === host2 &&
+    url1.pathname === url1.pathname &&
```

Comment:
we might need to make it lowercase before doing this comparison to avoid false positives due to the way Javascripts handles string casing